src/workflow/WorkflowDispatcher.js

```diff
@@ -180,21 +184,31 @@ export class WorkflowDispatcher {
     }
 
 
-    //this is used for testing
+    // this is used for testing.
     async addBatch(testData) {
         // check if testData is a function, and apply it to get the actual data
-        if (typeof testData === 'function') {
-            testData = await testData();
+        try {
+            if (typeof testData === 'function') {
+                testData = await testData();
+            }
+        } catch (error) {
+            console.error("Error executing testData function:", error);
+            throw error;
         }
+
         this.batchMode = true;
         if (Array.isArray(testData)) {
             this.batchCount += testData.length;
             for(let i=0;i<testData.length;i++){
-                await this.addToQueue(testData[i]);
+                if (Array.isArray(this.subscribeParams.acks) && this.subscribeParams.acks.includes(testData[i])) {
+                    console.debug(`Skipping already acknowledged test data: ${testData[i]}`);
+                } else {
+                    await this.addToQueue(testData[i]);
+                }
             }
         } else {
             this.batchCount += 1;
-            await this.addToQueue(testData);
+            await this.addToQueue(testData, ackFunctionGenerator(testData, this.subscribeParams));
```

Comment:
hmm, may be hard to understand what is allowed to go in the queue,. We don't have typings here but if you did use typescript it would be easier to tell what is allowed

Reply:
I've added docs to addToQueue for parameters, hopefully it helps in the interim between now and moving to TS.

You also caught a left over from a different approach, and is dead code for now. Removed  `ackFunctionGenerator` from the parameter here, we set testDataAckFunctionGenerator as part of the workflowDispatcher